frontend/src/hooks/useDIrector.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { ReactElement } from 'react'
+import useHandleYear from './useHandleYear'
+import useStudents from './useStudents'
+import { GetYearResponse, MutationOnLoad, ResponsePatch, StudentsGetResponse, UpdateStatusData } from '@/types'
+import { useMutation, UseMutationResult } from '@tanstack/react-query'
+import { ToastAction } from "@/components/ui/toast"
+import { toast } from "@/hooks/use-toast";
+
+type VariantToastProps = "default" | "success" | "destructive"
+
+const useDirector = () => {
+    const [initialSet , setInitialSet] = React.useState(false)
```

Comment:
import useState from react directment
```suggestion
    const [initialSet , setInitialSet] = useState(false)
```